### PR TITLE
Publish symbols to symweb

### PR DESCRIPTION
**Pull request by WilliamXieMSFT**
This publishes symbols to the Azure DevOps project associated with the build. For now, that build is internal to Microsoft. Once we have onboarded to the public dev.azure.com/ms/clrinstrumentationengine project, these symbols should be accessible by anyone.

### CODE REVIEW

File: build/yaml/jobs/windows/binaries.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 parameters:
   IndexSources: false
+  PublishSymbols: false
```
Comment (delmyers) [inline build/yaml/jobs/windows/binaries.yaml]: >PublishSymbols: false [](start = 2, length = 21)

Is there a pipeline that sets this to true? I don't see it in the code. If there isn't a pipeline that sets this, then we will never publish symbols.

Comment (WilliamXieMSFT) [reply build/yaml/jobs/windows/binaries.yaml]: Yes, this is set in the Jobs\binaries.yaml and is added for Signed Release builds

---
In reply to: [333118445](https://github.com/microsoft/CLRInstrumentationEngine/pull/134#discussion_r333118445) [](ancestors = 333118445)

File: build/yaml/jobs/binaries.yaml [hunk 2]
```diff
@@ -19,8 +19,9 @@ jobs:
       CopyFilters: |
         **
         !**\Nagler*
-        !**\*Tests*
-        !**\*Tests*\**
+        !**\*Test*
+        !**\*Test*\**
+      PublishSymbols: true
```
Comment (WilliamXieMSFT) [inline build/yaml/jobs/binaries.yaml]: >PublishSymbols: true [](start = 6, length = 20)

I'll look into adding this for Linux as well.

Comment (delmyers) [reply build/yaml/jobs/binaries.yaml]: Does this mean that we will be publishing symbols for all builds? Even PRs and CIs? Seems like we should only be doing it for the real signed build.

---
In reply to: [333181904](https://github.com/microsoft/CLRInstrumentationEngine/pull/134#discussion_r333181904) [](ancestors = 333181904)

Comment (WilliamXieMSFT) [reply build/yaml/jobs/binaries.yaml]: No, this is only for when SignType is set (to Test or Real). I can make it strictly for Real signed.

---
In reply to: [333184937](https://github.com/microsoft/CLRInstrumentationEngine/pull/134#discussion_r333184937) [](ancestors = 333184937,333181904)

Comment (delmyers) [reply build/yaml/jobs/binaries.yaml]: Didn't see the if statement above.

---
In reply to: [333184937](https://github.com/microsoft/CLRInstrumentationEngine/pull/134#discussion_r333184937) [](ancestors = 333184937,333181904)

Comment (WilliamXieMSFT) [reply build/yaml/jobs/binaries.yaml]: I'll add more comments here for clarity

---
In reply to: [333216207](https://github.com/microsoft/CLRInstrumentationEngine/pull/134#discussion_r333216207) [](ancestors = 333216207,333184937,333181904)

**Comment by WilliamXieMSFT** [comment]
>           Platform: 'x86'

We can build the name of each strategy based on these parameters.

---
Refers to: build/yaml/jobs/binaries.yaml:29 in fb2983e. [](commit_id = fb2983e2c382144305d342d428da8e6b7b385af0, deletion_comment = False)

**Comment by WilliamXieMSFT** [comment]
>   Configuration: Both

Remove both, so configuration will have to be passed. This will reduce the amount of duplicates in the matrix strategy parameter below.

---
Refers to: build/yaml/jobs/binaries.yaml:10 in fb2983e. [](commit_id = fb2983e2c382144305d342d428da8e6b7b385af0, deletion_comment = False)

File: build/yaml/jobs/binaries.yaml (hunk 2 above)
Comment (WilliamXieMSFT) [reply build/yaml/jobs/binaries.yaml]: Regarding the Linux symbols, I'll hold off as Aerobee doesn't do this either at the moment.

---
In reply to: [333242781](https://github.com/microsoft/CLRInstrumentationEngine/pull/134#discussion_r333242781) [](ancestors = 333242781,333216207,333184937,333181904)

**Pull request merged**